src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -117,6 +190,54 @@ public Uri GetEndpointForClient(ConfigurationClient client)
             ConfigurationClientWrapper currentClient = _clients.FirstOrDefault(c => c.Client == client);
             
             return currentClient?.Endpoint;
+        }
+
+        private async Task RefreshFallbackClients(CancellationToken cancellationToken)
+        {
+            IEnumerable<SrvRecord> results = await _srvLookupClient.Value.QueryAsync(_endpoint.DnsSafeHost, cancellationToken).ConfigureAwait(false);

```

Comment:
Any failures here shouldn't propagate outside of configurationclientmanager.

* Socket exception
* timeout
* srv record not found.

Reply:
Srv not found is not an exception.

I think the debatable problem is whether we need to break the user app's start-up if a socket exception/timeout is thrown when querying SRV DNS. My idea is, since we support auto failover by default, we should break it if there's problem with looking up.

Reply:
After think twice, I'm good with not throwing Socket exception timeout. We should not block the user's app startup on these cases.